### DISCUSSION TITLE
Display an image from the Resources in an AlertWindow popup with addCustomComponent()

### DISCUSSION THREAD

**Discussion by damiensellier** [RomanKubiak/ctrlr#247] [Q&A] [answered]
Hi,
I checked the Juce documentation about the AlertWindow:
[JUCE: AlertWindow Class Reference](https://docs.juce.com/master/classAlertWindow.html)

There's a function addCustomComponent()
component | the  component to add - _its size should be set up correctly before it is  passed in. The caller is responsible for deleting the component later on  - the AlertWindow won't delete it._

I want to display an image in my Credits Modal Box, since it's calling a component, I first added an uiImage "LogoModal" linked to the file imported in the Resources.  I had to place the uiImage outside the bounderies of the panel. Then,  I called the image component in the Modal Box : 

```
...
modalWindow = AlertWindow("Credits", "blah blah blah", AlertWindow.NoIcon)

CompLogoModal = panel:getModulatorByName("LogoModal"):getComponent()
CompLogoModal:setTopLeftPosition(50,100)
modalWindow:addCustomComponent(CompLogoModal)
...
```

The image is displayed but not at the right position.

1. Is there a way to create an image component (linked to the image in the resources pool) without creating a uiImage on the panel, just with LUA in the popup method?

2. Is there a way to set the position of the image with setTopLeftPosition(x,y) or something else, because the x,y coordinates doesn't change the position of the image in my case.

Thanks a lot for your help

Damien

**Comment by dnaldoog** [comment, marked as answer]
That is very interesting and thanks for this. I had no idea you could do that. 

This opens up a lot of possibilities.
 
 I managed to centre the image by placing it centred in a _uiCustomComponent_  using **g:drawImageAt** and placed the _uiComponent_ in a hidden layer, which you could do with _uiImage_ as well. 

<pre>
<code>
displayCredits = function()
    modalWindow = AlertWindow("Credits", "blah blah blah", AlertWindow.NoIcon)
    --modalWindow:addCustomComponent(CompLogoModal)
    modalWindow:addCustomComponent(custComp)
    modalWindow:addButton("    OK    ", 1, KeyPress(KeyPress.returnKey), KeyPress())
    modalWindow:addTextBlock("\169 2021")
    ret = modalWindow:runModalLoop()

    if ret == 1 then
    end

    modalWindow:setVisible(false)
end


paintComp = function(--[[ CtrlrComponent --]] comp, --[[ CtrlrComponent --]] g)

local w=comp:getWidth()
local h = comp:getHeight()
g:fillAll(Colour(0xff99ccff))

frResource=resources:getResource("logoModal")
if frResource~=nil then
  frImage =  frResource:asImage()
  g:drawImageAt(frImage,w/2-(frImage:getWidth()/2),h/2-(frImage:getHeight()/2),false)
end -- image found
g:setColour(Colour(0xffffff00):withAlpha(0.7))
g:fillRect(20,h-50,w-40,10)
end
</code>
</pre>
 
[uicomponent image in alertwindow_1_0_0_2021-04-05_09-26.zip](https://github.com/RomanKubiak/ctrlr/files/6255749/uicomponent.image.in.alertwindow_1_0_0_2021-04-05_09-26.zip)


![uicomp169](https://user-images.githubusercontent.com/76831683/113525656-31cb5c00-95f1-11eb-8dbd-cf2d529879e4.png)

**Reply by damiensellier** [reply]
@dnaldoog Absolutely, this addCustomComponent() feature is really nice.
Not all Juce class are available in Ctrlr but I was surprised this one was. It opens some door for complex popups and will improve our GUI. 
Thanks for the DEMO panel, this is exactly what I was talking about ;)

Not sure if we can customize the colors with colourIds() or setColour() : 
ColourIds { backgroundColourId = 0x1001800, textColourId = 0x1001810, outlineColourId = 0x1001820 }
That would be very nice.

Have a nice weekend
Damien

**Reply by Godlike-Productions** [reply]
Awesome, I was looking at a way of doing something similar, but worked around it.
